Separate out Separator from Setting control

jarolrod opened the pull request:
This is required in order to properly support the feature attempted by https://github.com/bitcoin-core/gui-qml/pull/214

### Master

| About | Dev | Prune | Connection | Node |
| ----- | --- | ----- | ---------- | ---- |
| <img width="752" alt="about-master" src="https://user-images.githubusercontent.com/23396902/218625414-180b5a16-6a54-4c30-9f7f-6fd7ba667a6d.png"> |<img width="752" alt="dev-master" src="https://user-images.githubusercontent.com/23396902/218629878-7a1b1024-5b25-446e-9159-6160f0699a8a.png"> |<img width="752" alt="prune-master" src="https://user-images.githubusercontent.com/23396902/218629926-f61080f2-e20a-42f8-8f1c-7dac9ec494a9.png"> |<img width="752" alt="connection-master" src="https://user-images.githubusercontent.com/23396902/218629963-baebb34c-9615-435a-9f7c-24a1a88c8706.png"> |<img width="752" alt="node-master" src="https://user-images.githubusercontent.com/23396902/218629984-1d6a2b5e-d4a4-4baf-ba28-041f90214036.png"> |

### PR 

| About | Dev | Prune | Connection | Node |
| ----- | --- | ----- | ---------- | ---- |
| <img width="752" alt="about-pr" src="https://user-images.githubusercontent.com/23396902/218625012-4fb45c2c-4911-422d-a83c-471e64b05fd1.png"> |<img width="752" alt="dev-pr" src="https://user-images.githubusercontent.com/23396902/218625047-78e6ca4d-d900-4bf7-9f69-77c1a40e0920.png"> |<img width="752" alt="prune-pr" src="https://user-images.githubusercontent.com/23396902/218625067-a71ee530-60c9-4c6f-86eb-f3dccac4230e.png"> |<img width="752" alt="connection-pr" src="https://user-images.githubusercontent.com/23396902/218625084-3811d101-c67b-4f8a-89e7-fc19d58ae609.png"> |<img width="752" alt="node-pr" src="https://user-images.githubusercontent.com/23396902/218625114-ec06b0ec-e37c-403c-a9c5-68b919375cba.png"> |


This also gets us in line with the design file in regards to the height of a setting
| master | pr | design |
| ------ | -- | ------ |
| <img width="483" alt="Screen Shot 2023-02-13 at 7 30 58 PM" src="https://user-images.githubusercontent.com/23396902/218630233-cb861aad-9541-419a-9252-6fbbe697bad8.png"> | <img width="483" alt="Screen Shot 2023-02-13 at 7 21 49 PM" src="https://user-images.githubusercontent.com/23396902/218630255-e428a704-9b8c-41ee-a065-ebef73a9f75e.png"> | <img width="483" alt="Screen Shot 2023-02-13 at 7 31 54 PM" src="https://user-images.githubusercontent.com/23396902/218630286-6bb415a7-eca6-44d3-8d12-faf5dfe283b7.png"> |

[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/263)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/263)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/263)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/263)

johnny9 commented on src/qml/components/Separator.qml:
```diff
@@ -0,0 +1,12 @@
+Rectangle {
+    height: 1
+    color: Theme.color.neutral5
+}
```
github complaining about no newline at eof

jarolrod replied:
thanks, addressed in 5cbf67f1c479f08cfe3954c56c9962029e6bd5aa

jarolrod commented:
Updated from from [1ea3922 ](https://github.com/bitcoin-core/gui-qml/commit/1ea392243526ead39d0c660ff54f6886703edd39)to [5cbf67f](https://github.com/bitcoin-core/gui-qml/commit/5cbf67f1c479f08cfe3954c56c9962029e6bd5aa), [compare](https://github.com/bitcoin-core/gui-qml/compare/1ea392243526ead39d0c660ff54f6886703edd39..5cbf67f1c479f08cfe3954c56c9962029e6bd5aa)

changes: addressed no newline at eof

johnny9 reviewed (approved):
ACK 5cbf67f1c479f08cfe3954c56c9962029e6bd5aa

Good change, will rebase the Peers layout on this
